Source/SelfService/Web/microservice/dataTable.tsx

```diff
@@ -165,7 +175,7 @@ export const DataTable = ({ application, environment, microservices }: DataTable
             flex: 1,
             renderCell: statusCell,
             sortComparator: (v1, v2, param1, param2) =>
-                param1.value[0].phase.localeCompare(param2.value[0].phase)
+                param1.value[0]?.phase?.localeCompare(param2.value[0]?.phase)
```

Comment:
I believe this function might return `undefined` - which can cause undefined behaviour. So I suggest that we change this a little bit.

First, we can define a set of statuses that we care about - and then return them from the backend status:
```typescript
enum MicroserviceStatus {
    Running = 0,
    Pending = 1,
    Failing = 2,
    Unknown = 3,
};

const getMicroserviceState = (phase?: string): MicroserviceStatus => {
    const checkStatus = phase?.toLowerCase?.();

    if (typeof checkStatus !== 'string') {
        return MicroserviceStatus.Unknown;
    } else if (checkStatus.includes('running')) {
        return MicroserviceStatus.Running;
    } else if (checkStatus.includes('pending')) {
        return MicroserviceStatus.Pending;
    } else if (checkStatus.includes('failed')) {
        return MicroserviceStatus.Failing;
    }

    return MicroserviceStatus.Unknown;
}
```

Then we can change the `statusCell` to use that status for rendering:
```typescript

const statusCell = (params: GridRenderCellParams) => {
    let color = themeDark.palette.text.primary;
    let icon = <QuestionMark sx={{ color }} />;

    switch (getMicroserviceState(params.row.status[0]?.phase)) {
        case MicroserviceStatus.Running:
            icon = <CheckCircleRounded />;
            break;
        case MicroserviceStatus.Pending:
            color = themeDark.palette.warning.main;
            icon = <WarningRounded sx={{ color }} />;
            break;
        case MicroserviceStatus.Failing:
            color = themeDark.palette.error.main;
            icon = <ErrorRounded sx={{ color }} />;
            break;
        case MicroserviceStatus.Unknown:
            return 'N/A';
    }

    return (
        <Box sx={styles.status}>
            {icon}
            <Typography sx={{ ...styles.statusTitle, color }}>{status}</Typography>
        </Box>
    );
}
```

Then, lastly - we can change this sortComparator to sort using the numeric value of that enum we created:
```typescript
            sortComparator: (_, __, left, right) => {
                const leftStatus = getMicroserviceState(left.value[0]?.phase);
                const rightStatus = getMicroserviceState(right.value[0]?.phase);
                return leftStatus - rightStatus;
            }
```

This should result in a stable sorting. If you want to change the order of which things are sorted (maybe we want the errors on top?) - you can just rearrange the numbers in this enum:
```typescript
enum MicroserviceStatus {
    Running = 0,
    Pending = 1,
    Failing = 2,
    Unknown = 3,
};
```